src/parser/metrics/RealLinesOfCode.ts

```diff
@@ -32,10 +33,15 @@ export class RealLinesOfCode implements Metric {
      * this is the most efficient way to traverse a syntax tree.
      * @param cursor A {@link TreeCursor} for the syntax tree.
      * @param realLinesOfCode A set in which the line numbers of the found code lines are stored.
+     * @param isComment Function that checks whether a node is a comment or not.
      */
-    walkTree(cursor: TreeCursor, realLinesOfCode = new Set<number>()) {
+    walkTree(
```

Comment:
i suggest to not pass a lambda function as parameter to make the code easy to understand. Let's write a function to prove if the syntax node is a comment in a specific language. That approach may lead to minimal increased runtime, but it spares time in development for later.

Reply:
I tend to disagree, as having the distinction of cases for the programming languages at the beginning of the calculate()-function makes it more obvious that there is a special treatment for a language here and where to add a new special treatment later. Also I don't like making things less efficient, yes. And it should not be too hard to understand what is going on here.